core/src/Streamly/Internal/Data/Array/Foreign/Mut/Type.hs

```diff
@@ -2353,12 +2352,12 @@ bubbleAsc cmp0 arr =
 
         l = length arr
 
-        go x i =
+        go x i = do
             if i >= 0
             then do
                 x1 <- getIndexUnsafe i arr
-                case x1 `cmp0` x of
-                    GT -> do
+                case x `cmp0` x1 of
+                    LT -> do
```

Comment:
Is this changes required? Was the original code incorrect?

Reply:
no, no